packages/client/src/create-document.tsx

```diff
@@ -1,42 +1,27 @@
 import React from "react";
-import styled from "@emotion/styled";
 import { useDispatch, useSelector } from "react-redux";
 import { selectUser } from "./redux/appState/user";
 import { addDocument } from "./redux/appState/document";
-
-const Form = styled.form`
-  max-width: 300px;
-  display: flex;
-  flex-direction: column;
-`;
+import { usePostDocumentMutation } from "./redux/api";
 
 export default function CreateDocument() {
   const me = useSelector(selectUser);
-  const [posting, setPosting] = React.useState(null);
+  const [postDocument, { isLoading }] = usePostDocumentMutation();
   const dispatch = useDispatch();
 
+  const handleCreate = async () => {
+    const doc = await postDocument().unwrap();
+    dispatch(addDocument(doc));
+  };
+
   return (
-    <Form
-      onSubmit={(e) => {
-        e.preventDefault();
-        setPosting(true);
-        fetch("/api/document", {
-          method: "POST",
-        })
-          .then((data) => data.json())
-          .then(({ document }) => {
-            if (!document.id || !document.creator_id) return;
-            dispatch(addDocument(document));
-            setPosting(false);
-          });
-      }}
-    >
+    <div>
```

Comment:
Better to avoid making style changes and functional changes at the same time. 

Can we keep the dispatch inside the form's `onSubmit` function?

Reply:
I'll switch back to Form, not really sure why I made that change to begin with tbh